### ISSUE TITLE
Imagenet classification with deep convolutional neural networks.

### ISSUE THREAD

**Issue by NorihiroNakaoka** [NorihiroNakaoka/papers#8]
## 0. 論文
[Imagenet classification with deep convolutional neural networks.](http://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf)
Krizhevsky, Alex, Ilya Sutskever, and Geoffrey E. Hinton.
## 1. どんなもの？
 ImageNet LSVRC-2010コンテストの120万画素のハイレゾ画像を1000種類に分類するためのニューラルネット
error rateはtop1が37.5% top5が 17.0%．学習速度を速くするためにnon-saturating(ReLUなど，sigmoidやtanhではない) neuronsと畳み込み演算の効率的なGPUの実装，過学習を減らすためにdropoutを使用．この改変版モデルでILSVRC-2012で優勝，top-5エラー率が15.3%と二位の26.2%を大きく引き離した．

### 結果
ILSVRC-2010でエラー率Top1では17.0%, Top5では37.5%とを達成．
<img width="234" alt="screenshot 2018-07-14 21 07 18" src="https://user-images.githubusercontent.com/16438057/42724262-059aef04-87aa-11e8-8e83-40295063b61a.png">

## 2. 先行研究と比べてどこがすごい？
ImageNetでの画像認識コンペティションでかつての手法をdeep learningにより淘汰

### 構造
* 8つの学習層
* 5つの畳み込み層
* 3つの全結合層
<img width="559" alt="screenshot 2018-07-15 00 51 39" src="https://user-images.githubusercontent.com/16438057/42726068-41e362f6-87c9-11e8-98b1-5a89bb5509ed.png">

#### 3.1.ReLU Nonlinearity
通常活性化関数にはf(x) = tanh(x) or f(x) = (1 + e−x)−1（saturating func）などを用いるが，
CNNでReLUを用いることで何倍も早く学習可能．
大きいモデルやデータセットで学習するのには早く学習をすることが重要．

#### 3.2.Training on Multiple GPUs
 GTX 580 GPUのメモリは3GBしかなく，それがネットワークの最大サイズの限界になる．
このプロジェクトでは2つのGPUを使用．現在のGPUは直接他のGPUのメモリにデータの書き込みができる．
結果として生じる構造は円柱とどこか似ている．（図2）
このスキームはGPU一つで半分のカーネルで学習したものと比べて，top1のエラー率を1.7%,top5のエラー率を1.２%減らした．

#### 3.3.Local Response Normalization
ReLUはSaturationを防ぐために入力の正規化を必要としない説明変数を持っている．
しかし，学習を助けるために以下の数式が使える．
<img width="334" alt="screenshot 2018-07-24 16 30 49" src="https://user-images.githubusercontent.com/16438057/43123396-fe3150ca-8f5e-11e8-9e8c-440d5e80f66c.png">
最適なパラメータは以下
k = 2, n = 5, α = 10<sup>−4</sup>, and β = 0.75
これでエラー率がTop1で1.4%，Top5で1.2%向上．

#### 3.4 Overlapping Pooling
普通CNNのプーリングはOverlap(重複)しない．
z × zのプーリングユニットをsずつ移動させるとする．s = zとすると普通のプーリング, s < zとするとOverlapping poolingとなる．
論文中ではs = 2, z=3とした時，s=2, z=3 とした時と比べてTop1のエラー率が0.4%,Top5が0.3%向上した．
あとOverlapping poolingは少しだけ過学習になりにくい．

#### 3.5 Overall Architecture
5つの畳み込み層と3つの全結合層
最後の全結合層は1000クラスのラベル上に分類する1000通りのsoftmaxにより供給される
多項ロジステック回帰を最大化＝予想分布のもとで正しいラベルの対数確率の複数訓練ケースにわたる平均の最大化
2,3,5個目のカーネルは同じGPUの中にある一つ前の層のカーネルマップにだけつながっている
三個目の畳み込み層は2つ目のレイヤーの全てのカーネルマップに繋がっている
全結合層のニューロンはすべての前の層のすべてのニューロンに繋がっている
ReLUはすべての畳み込み層と全結合層の出力に使われる

最初の畳み込み層のフィルターは224 × 224 × 3の入力イメージを4ピクセルずつずらしながら 11 × 11 × 3のサイズの96個のカーネルでフィルターする

二つ目の畳み込み層は応答正規化とプーリングされた入力（一つ目の畳み込み層の出力）を256個のsize 5 × 5 × 48でフィルターする

3,4,5個目の畳み込み層は正規化とプーリング無しに接続される
3つ目の畳み込み層は384個のカーネルで3 × 3 × 256サイズを持つ．2つ目の畳み込み層の出力と繋がっている
4つ目の畳み込み層は384 kernels of size 3 × 3 × 192
5つ目の畳み込み層は256 kernels of size 3 × 3 × 192

全結合層は4096のニューロンを持つ．

### 4 Reducing Overfitting
このネットワークは6億のパラメータを持つ．ILSVRCの1000クラス分類はイメージをラベルにマッピングするのに10bitの制約をそれぞれのトレーニングに課すが，これでは過学習を防ぐのに足りない．
過学習を撲滅するために主に以下の二つを行う
* Data Augmentation
* Dropout

#### 4.1 Data Augmentation
label-preserving transformationsを使ってデータセットを増やす．計算量は少ないのでCPUで処理，その間に前のバッチをGPUで学習

##### image translations and horizontal reflec- tions
1つめの方法
256×256のイメージから224 × 224のパッチをランダムに抽出
これによりトレーニングセットのサイズが2048倍になる
これをやると大きなネットワークでも過学習を起こさずに学習できる
5つの224×224のパッチ（4つのコーナーパッチと中央パッチ）とそれらの水平反射（したがって10つのパッチ）を抽出し、ネットワークのソフトマックスレイヤーによる予測を平均化することによって予測を行います

##### altering the intensities of the RGB channels in training images
2つめの方法
具体的には、ImageNetトレーニングセット全体のRGBピクセル値のセットに対してPCAを実行します。
各訓練画像に、見つかった主成分の倍数を足す
主成分は平均ゼロおよび標準偏差0.1のガウス関数からから引かれた確率変数× 対応する固有値に比例する大きさ．
<img width="48" alt="screenshot 2018-07-26 03 34 43" src="https://user-images.githubusercontent.com/16438057/43220418-ef8b7fc4-9084-11e8-96af-1370e44200ea.png">
<img width="103" alt="screenshot 2018-07-26 03 34 38" src="https://user-images.githubusercontent.com/16438057/43220422-f1e60b18-9084-11e8-8f8d-1a3012d7c367.png">
に
<img width="214" alt="screenshot 2018-07-26 03 32 43" src="https://user-images.githubusercontent.com/16438057/43220427-f533b5ea-9084-11e8-88f1-591958575809.png">
を加える．

## 4. どうやって有効だと検証した？
ImageNet LSVRC-2010コンテストで優勝してる

## 6. 次に読むべき論文は？
